src/bp-core/profile-search/bps-xprofile.php

```diff
@@ -460,7 +460,12 @@ function bp_ps_xprofile_gender_users_search( $f ) {
 
 		$exists_gender = $wpdb->get_results( "SELECT COUNT(*) as count, id FROM {$table_profile_fields} a WHERE parent_id = 0 AND type = 'gender' " );
 
-		$custom_ids = $wpdb->get_col( "SELECT user_id FROM {$table_profile_data} WHERE field_id = {$exists_gender[0]->id} AND value = '{$gender}'" );
+		$where = " AND value = '{$gender}'";
+		if ( is_array( $gender ) ) {
+			$gender = "'" . implode( "', '", $gender ) . "'";
+			$where = " AND value IN({$gender})";
+		}
+		$custom_ids = $wpdb->get_col( "SELECT user_id FROM {$table_profile_data} WHERE field_id = {$exists_gender[0]->id} {$where}" );
```

Comment:
@mahedihasannoman Please make it condtional that `$exists_gender` is not empty otherwise `$exists_gender[0]->id` gives fatal.

```suggestion
		if ( ! empty( $exists_gender ) ) {
			$where = " AND value = '{$gender}'";
			if ( is_array( $gender ) ) {
				$gender = "'" . implode( "', '", $gender ) . "'";
				$where  = " AND value IN({$gender})";
			}
			$custom_ids = $wpdb->get_col( "SELECT DISTINCT user_id FROM {$table_profile_data} WHERE field_id = {$exists_gender[0]->id} {$where}" );

			if ( ! empty( $custom_ids ) ) {
				return $custom_ids;
			} else {
				return array();
			}
		} 
```

Reply:
Done